### PR TITLE
Implement the sync-agent-groups-get command

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#11005|

## Description
This PR implements the command in charge of acquiring the agents to be synchronized and all the groups belonghip for each agent.
The response is formatted in a JSON structures array, where each element is complete and the response data is fully parseable.
If a response doesn´t fit in the socket size, it is chopped in chunks, returning a **due** status that represents that more data is pending to be delivered.
The command expects a json structure with the last agent id to continue obtaining info in case the response is sent by chunks and the synchronization condition:
```
global sync-agent-groups-get {"last_id":0, "sync_condition":"sync_status"}
global sync-agent-groups-get {"last_id":0, "sync_condition":"cks_mismatch"}
```
The response of the command has the following format, meant to be used in the **set** command directly:
`ok [{"id":1,"groups":["TEST_GROUP1","TEST_GROUP2"]},{"id":2,"groups":["TEST_GROUP1","TEST_GROUP2"]}]`

## Tests
For this task, a script capable to execute multiple commands and verify the response was implemented and attached in the Epic QA task (https://github.com/wazuh/wazuh/issues/11240#issuecomment-992999551)
Several commands where executed to confirm the correct behavior of the command and the format of the response.

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Comment by palaciosjeremias** [comment]
Scan build results:
image

**Comment by palaciosjeremias** [comment]
## Valgrind results:
image

**Comment by palaciosjeremias** [comment]
## WDB_tester tests results:
[test_results.txt](https://github.com/wazuh/wazuh/files/7707882/test_results.txt)

**Review by DProvinciani — COMMENTED** [review]
Just a minor comment.

File: src/wazuh_db/schema_global.sql [hunk 1]
```diff
@@ -66,7 +68,7 @@ CREATE TABLE IF NOT EXISTS `group` (
 
 CREATE TABLE IF NOT EXISTS belongs (
     id_agent INTEGER REFERENCES agent (id) ON DELETE CASCADE,
-    id_group INTEGER REFERENCES `group` (id) ON DELETE CASCADE,    
+    id_group INTEGER,
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/wazuh_db/schema_global.sql:71]: You are right. Now it's eliminated on both files.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**